crypto/mbedtls/apps/selftest/Makefile

```diff
@@ -0,0 +1,33 @@
+############################################################################
+# apps/crypto/mbedtls/apps/selftest/Makefile
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+include $(APPDIR)/Make.defs
+include ../Makefile
+
+# mbedtls built-in benchmark application info
+
+PROGNAME  = "mbedselftest"
+PRIORITY  = $(CONFIG_MBEDTLS_APP_SELFTEST_PRIORITY)
+STACKSIZE = $(CONFIG_MBEDTLS_APP_SELFTEST_STACKSIZE)
+MODULE    = $(CONFIG_MBEDTLS_APP_SELFTEST)
+
+MAINSRC = $(MBEDPROGDIR)/test/selftest.c
```

Comment:
why not use the relative path? So we can remove MBEDPROGDIR and  crypto/mbedtls/apps/Makefile.

Reply:
I could not get that to work, but maybe I am doing something wrong
with:
```
MAINSRC = ../../mbedtls/test/selftest.c
```
I get this
```
make[3]: *** No rule to make target '../../mbedtls/test/selftest.c', needed by '../../mbedtls/test/selftest.home.bashton.nuttx.wrk.apps.crypto.mbedtls.apps.selftest.o'.  Stop.
```

Reply:
I know your problem, both selftest.c and benchmark.c put in crypto/mbedtls/programs/test, but you put Makefile under:
crypto/mbedtls/apps/selftest/Makefile
crypto/mbedtls/apps/benchmark/Makefile
Why not put Makfile to:
crypto/mbedtls/programs
and merge two apps into one Mafile?

Reply:
You cannot call `include $(APPDIR)/Application.mk` twice in the same make file which is they need to be in there own folders.  But I don't think that is related to this path issue.

Reply:
also careful with your paths.
`crypto/mbedtls/programs/test` does not exist. `crypto/mbedtls/mbedtls/programs/test` does, but that is inside the downloaded source.

Reply:
but you can specify multiple apps in PROGNAME and MAINSRC like zmodem:
https://github.com/apache/incubator-nuttx-apps/blob/master/system/zmodem/Makefile#L38
https://github.com/apache/incubator-nuttx-apps/blob/master/system/zmodem/Makefile#L45

Reply:
Hmm interesting I did not realize that.  Can you take a look at how I set it up now.  MODULE only take a single value so I had to create a wrapper for all the apps to drive that, but I think it is reasonable.

Reply:
Yes, another normal approach is reuse CRYPTO_MBEDTLS option.